core/dispatch.c

```diff
@@ -714,7 +714,7 @@ dispatch_enter_native(dcontext_t *dcontext)
          * be an alarm signal so we drop it as the simplest solution.
          */
         ASSERT(dcontext->signals_pending);
```

Comment:
I wonder if usages like this need to be audited too. `signals_pending` can be < 0 if one is being currently handled.

Reply:
Done.  This one is >0; all others are !=0.